src/components/DefaultSidebar.test.tsx

```diff
@@ -0,0 +1,270 @@
+      // -----------------------------------------------------------------------
+
+      {
+        act(() => {
+          _setDocked(undefined);
+        });
+
+        expect(h.state.defaultSidebarDockedPreference).toBe(false);
+
+        const dockButton = await waitFor(() => {
+          const sidebar = container.querySelector<HTMLElement>(".test-sidebar");
+          expect(sidebar).not.toBe(null);
+          const dockButton = queryByTestId(sidebar!, "sidebar-dock");
+          expect(dockButton).not.toBe(null);
+          return dockButton!;
+        });
+
+        fireEvent.click(dockButton);
+        await waitFor(() => {
+          expect(h.state.defaultSidebarDockedPreference).toBe(true);
+          expect(dockButton).toHaveClass("selected");
+        });
+
+        fireEvent.click(dockButton);
+        await waitFor(() => {
+          expect(h.state.defaultSidebarDockedPreference).toBe(false);
+          expect(dockButton).not.toHaveClass("selected");
+        });
+      }
+    });
+  });
+
+  it("DefaultSidebar — dockable behavior", async () => {
+    let _setOnDock: (
+      value: undefined | false | (() => void),
+    ) => void | undefined = null!;
+    let _setDocked: (docked: boolean) => void = null!;
+
+    const CustomExcalidraw = () => {
+      const [docked, setDocked] = React.useState<boolean | undefined>();
+      _setDocked = setDocked;
+      const [onDock, setOnDock] = React.useState<
+        (() => void) | false | undefined
+      >();
+      _setOnDock = setOnDock;
+      return (
+        <Excalidraw
+          initialData={{
+            appState: { openSidebar: { name: DEFAULT_SIDEBAR.name } },
+          }}
+        >
+          <DefaultSidebar
+            className="test-sidebar"
+            docked={docked}
+            onDock={onDock}
+          >
+            hello
+          </DefaultSidebar>
+        </Excalidraw>
+      );
+    };
+
+    await render(<CustomExcalidraw />);
+
+    const { h } = window;
+
+    // if `docked={undefined}` & `onDock={undefined}`, should allow docking
+    // as normal
+    // -------------------------------------------------------------------------
+    await withExcalidrawDimensions({ width: 1920, height: 1080 }, async () => {
+      expect(h.state.defaultSidebarDockedPreference).toBe(false);
+
+      const { dockButton } = await assertSidebarDockButton(true);
+
+      fireEvent.click(dockButton);
+      await waitFor(() => {
+        expect(h.state.defaultSidebarDockedPreference).toBe(true);
+        expect(dockButton).toHaveClass("selected");
+      });
+
+      fireEvent.click(dockButton);
+      await waitFor(() => {
+        expect(h.state.defaultSidebarDockedPreference).toBe(false);
+        expect(dockButton).not.toHaveClass("selected");
+      });
+    });
+
+    // if `docked={undefined}` & `onDock={() => {}}`, should allow docking
+    // as normal (handled by editor), and only listen on changes
```

Comment:
instead of having these comments here , can you instead split the test ?
1. when `docked={undefined}` & `onDock={undefined}`, should allow docking
2. when `docked={undefined}` & `onDock={() => {}}`, should allow docking
3. when`onDock={false}`, should disable docking
4. when `docked={true}` & `onDock={undefined}`, should disable docking for docked sidebar
5. when `docked={false}` & `onDock={undefined}`, should disable docking and force sidebar to be undocked

Right now its in a single test which really impacts readability so breaking it will simplify things a lot and help in better readability.

Same for the previous test as well.